plugins/clerk/steps/delete-user.ts

```diff
@@ -0,0 +1,93 @@
+import "server-only";
+
+import { fetchCredentials } from "@/lib/credential-fetcher";
+import { type StepInput, withStepLogging } from "@/lib/steps/step-handler";
+import { getErrorMessage } from "@/lib/utils";
+import type { ClerkCredentials } from "../credentials";
+
+type DeleteUserResult =
+  | { success: true; deleted: boolean }
+  | { success: false; error: string };
+
+export type ClerkDeleteUserCoreInput = {
+  userId: string;
+};
+
+export type ClerkDeleteUserInput = StepInput &
+  ClerkDeleteUserCoreInput & {
+    integrationId?: string;
+  };
+
+/**
+ * Core logic - portable between app and export
+ */
+async function stepHandler(
+  input: ClerkDeleteUserCoreInput,
+  credentials: ClerkCredentials
+): Promise<DeleteUserResult> {
+  const secretKey = credentials.CLERK_SECRET_KEY;
+
+  if (!secretKey) {
+    return {
+      success: false,
+      error:
+        "CLERK_SECRET_KEY is not configured. Please add it in Project Integrations.",
+    };
+  }
+
+  if (!input.userId) {
+    return {
+      success: false,
+      error: "User ID is required.",
+    };
+  }
+
+  try {
+    const response = await fetch(
+      `https://api.clerk.com/v1/users/${input.userId}`,
```

Comment:
In get-user.ts, update-user.ts, and delete-user.ts, user-controlled input is directly interpolated into URL paths without sanitization.

If input.userId contains malicious content like:
- ../../other-endpoint (path traversal)
- user123?admin=true (query string injection)
- URL-encoded special characters

The request could be manipulated.

Mitigation Recommendation: Add URL encoding:

```
const response = await fetch(
  `https://api.clerk.com/v1/users/${encodeURIComponent(input.userId)}`,
```